packages/common/core-interfaces/src/jsonable.ts

```diff
@@ -78,16 +79,17 @@ export interface Internal_InterfaceOfJsonableTypesWith<T> {
  * ```typescript
  * function foo<T>(value: Jsonable<T>) { ... }
  * ```
- * @alpha
+ * @public
  */
-export type Jsonable<T, TReplaced = never> = /* test for 'any' */ boolean extends (
+export type Jsonable<T = unknown, TReplaced = never> = /* test for 'any' */ boolean extends (
```

Comment:
Yes. I am only in favor of 1 & 2 after improvement. Or we expose a replacement public guy. We are also missing the deserialized form. See my earlier review comments and also https://github.com/jason-ha/FluidFramework/blob/lcedds-proto/experimental/framework/independent-state/src/jsonDeserialized.ts

Reply:
Your direction is completely correct, and this was more or less in my Distributed Independent State backlog. Overall, it has been on my mind as well, ever since I started looking at signal support. I think the event typing should be improved as well.
I was very surprised and later happy that Jsonable is not public. For a while it was miraculously internal. I became happy because we should break it. It is not restrictive enough. For example, it should not allow `undefined`. Those won't come through.
I think we have two options:
1. Introduce a clean new type perhaps called `JsonEncodable` into core-interfaces. This can be the version I have been working on or something even better - no one has really seen it. Ideally, uses of `Jsonable` would move to adopt the new type filter. Where public it will be breaking as now more restrictive. This is uncoupled and unblocked.
2. Just replace `Jsonable` with the changes. Deal with all of the fallout in repo and likely for Loop.

I put out a draft to show what 2 looks like and it is also a good way to compare the two: #20968 

The other thing we need in this space is the output version Jsonable. They are not identical. In many cases, I think was also want to make things `readonly`.

Reply:
Here is my next step proposal: #21725